unstructured/partition/docx.py

```diff
@@ -287,6 +287,10 @@ def _iter_document_elements(self) -> Iterator[Element]:
         # -- functions like `._iter_paragraph_elements()` where the "just return when done"
         # -- characteristic of a generator avoids repeated code to form interim results into lists.
 
+        if not self._document.sections:
+            for paragraph in self._document.paragraphs:
+                yield from self._iter_paragraph_elements(paragraph)
```

Comment:
This will work in this case, but not the general case, because a document can contain both paragrahs and tables and tables are of course skipped by `Document.paragraphs`.

If we're looking for a quick-fix for the Microsoft Teams chat transcript problem then I'd say this is an entirely valid approach. But longer term we probably want a solution that includes the (real) possibility of tables being in a "sectionless" document.

Offhand I'd expect tables are unlikely in a Teams chat transcript, but I've never used Teams or its chat and I suppose even if a user can't enter a table in chat, certain chat events (like I don't know, maybe multi-person joins or leaves or something) could conceivably be _rendered_ as a table when the chat is exported as DOCX.

I can sketch out a more robust solution for longer-term use and put that in a follow-up PR if we want to let this one go to get us around the immediate problem.

Reply:
hmm, do we have a way to know what possible iterables are in a docx object? Cause the impulse here is then "ok, lets add `self._document.other_iterables` to the loop". But then I guess the questions is are there documented way to access a "root" iterable? Or maybe reverse it and put everything inside a section upon loading a docx?

Reply:
Yeah, I have a sketch almost done I can show you. Essentially this is the gist:

A Word document (the main document, not including headers footers, other bits like notes, footnotes, comments, etc.) can contain _block-items_, which are either paragraphs or tables.

For historical reasons, `python-docx` has `Document.paragraphs` and `Document.tables`. If we use both of those we get everything. Problem is the sequence is lost, like all paragraphs appear before all tables or vice-versa.

The way to go is to iterate through the `<w:p>` and `<w:tbl>` elements in the document body and then create a `Paragraph` or `Table` object as appropriate. This produces them in document order.

That's what I'm sketching out now.